src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java

```diff
@@ -12,21 +15,62 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Message create(Message entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Failed to save message" + entity, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Message get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return Optional.ofNullable(session.get(Message.class, id)).orElse(null);
```

Comment:
```suggestion
            session.get(Message.class, id);
```

i would recommend changing your Optional.ofNullable to session.get(Class.class, id); in all of your get() methods
I don't see why you'd need Optional here